File: SortAlgorithms/insertion_sort.py
#!/usr/bin/env python

# Created by Michael Gilliland
# Date: Thu Jan  3 18:02:47 EST 2013
# 
# 

def insertion_sort(list):
    for i in xrange(1, len(list)):
        k = i
        item = list[i]
        while k > 0 and item < list[k-1]:
            list[k] = list[k-1]
            k -= 1
        list[k] = item

if __name__ == '__main__':
    sort_me = [1, 43, 123, 5, 2, 654, 23, 4, 123]
    insertion_sort(sort_me)
    print sort_me

    sort_me = [134, 21, "Michael", 24, 1, "Foobar", 'foobar', "Django Unchained", 23, 12, 1, "j"]
    insertion_sort(sort_me)
    print sort_me

File: SortAlgorithms/insertionsort.cpp
#include <iostream>
using namespace std;

void insertionsort(int arr [], int size);

int main()
{
	int a[12] = {123, 1, 235, 123, 1, 34, 2, 656, 22, 54, 27, 27};

	int b[1] = {4};

	insertionsort(a, 12);
        insertionsort(b, 1);

	for (int i = 0; i < 12; i++)
		cout << a[i] << ", ";

	cout << endl;

	for (int i = 0; i < 1; i++)
		cout << b[i] << ", ";

	cout << endl;

	return 0;
}

void insertionsort(int arr [], int size)
{
        if (size < 2)
        	return;

	for (int i = 1; i < size; i++)
	{
		int index = i, item = arr[index];

		for (index = i; index > 0 && item < arr[index-1]; index--)
			arr[index] = arr[index-1];

		arr[index] = item;
	}
}
